src/transformers/tokenization_tapas.py

```diff
@@ -0,0 +1,1194 @@
+# coding=utf-8
+# Copyright 2020 (...) and The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Tokenization class for TAPAS model."""
+
+
+import ast
+import collections
+from dataclasses import dataclass
+from typing import Dict, List, Optional, Text, Tuple, Union
+
+import torch
+
+from transformers import tokenization_tapas_utilities as utils
+
+from .tokenization_bert import BertTokenizer
+from .tokenization_utils_base import (
+    ENCODE_KWARGS_DOCSTRING,
+    ENCODE_PLUS_ADDITIONAL_KWARGS_DOCSTRING,
+    INIT_TOKENIZER_DOCSTRING,
+    AddedToken,
+    BatchEncoding,
+    EncodedInput,
+    EncodedInputPair,
+    PaddingStrategy,
+    PreTokenizedInput,
+    PreTokenizedInputPair,
+    PreTrainedTokenizerBase,
+    TensorType,
+    TextInput,
+    TextInputPair,
+    TruncationStrategy,
+)
+
+
+VOCAB_FILES_NAMES = {"vocab_file": "vocab.txt"}
+
+PRETRAINED_VOCAB_FILES_MAP = {
+    "vocab_file": {
+        # to be added
+    }
+}
+
+PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES = {
+    # to be added
+}
+
+
+PRETRAINED_INIT_CONFIGURATION = {
+    # to be added
+}
+
+
+@dataclass(frozen=True)
+class TokenCoordinates:
+    column_index: int
+    row_index: int
+    token_index: int
+
+
+@dataclass
+class TokenizedTable:
+    rows: List[List[List[Text]]]
+    selected_tokens: List[TokenCoordinates]
+
+
+@dataclass(frozen=True)
+class SerializedExample:
+    tokens: List[Text]
+    column_ids: List[int]
+    row_ids: List[int]
+    segment_ids: List[int]
+
+
+def _is_inner_wordpiece(token: Text):
+    return token.startswith("##")
+
+
+class TapasTokenizer(BertTokenizer):
```

Comment:
I don't think this tokenizer needs to inherit from the `BertTokenizer`, as it behaves very differently.